### PR TITLE
Feature/landing screen

### PR DESCRIPTION
**Pull request by apptonio**

### CODE REVIEW

File: lib/screens/home/home_screen.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import 'package:apexology/screens/landing/landing_controller.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+
+//import 'home_controller.dart';
+
+class HomeScreen extends StatelessWidget {
```
Comment (PetricMatej) [inline lib/screens/home/home_screen.dart]: we are using getX so replace this line with class HomeScreen extends GetView<HomeController> {

File: lib/screens/home/home_screen.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  const HomeScreen({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    //final homeController = Get.put(HomeController());
```
Comment (PetricMatej) [inline lib/screens/home/home_screen.dart]: delete this line, home controller can be used like: controller.somefunction

File: lib/constants/colors.dart [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
-/// You can then reference them in code with `MyColors.someColor`
-///
-
 class MyColors {
   static const white = Color(0xFFFFFFFF);
   static const black = Color(0xFF000000);
   static const red = Color(0xFFFF0000);
   static const green = Color(0xFF00FF00);
   static const blue = Color(0xFF0000FF);
+  static const grey = Color.fromRGBO(30, 39, 46, 1.0);
```
Comment (lukacevica) [inline lib/constants/colors.dart]: Try to be consistent. Use either one or the other way for colors.

File: lib/constants/localization.dart [hunk 4]
```diff
@@ -20,13 +12,38 @@ class Localization extends Translations {
...
+    'disabledEmail': 'The email for this account has been disabled.',
+    'invalidEmail': 'Please enter a valid email address.',
+    'formError': 'An error occurred, please check your credentials.',
+    'betterCredentials': 'Please provide a better email & password.',
+    'weakPassword': 'Please provide a stronger password.',
+    'emailAlreadyInUse': 'An account with this email already exists.',
   };
 
-  /// Croatian strings
   final hr = {
```
Comment (lukacevica) [inline lib/constants/localization.dart]: Each language should be in separate file.

File: lib/constants/localization.dart [hunk 5]
```diff
@@ -20,13 +12,38 @@ class Localization extends Translations {
         'hr': hr,
       };
 
-  /// English strings
   final en = {
```
Comment (lukacevica) [inline lib/constants/localization.dart]: Separate the words into groups for each screen/module respectively.

File: lib/constants/text_styles.dart [hunk 6]
```diff
@@ -1,23 +1,56 @@
...
+    color: MyColors.blue,
+    fontSize: 12.sp,
+    fontWeight: FontWeight.w500,
+  );
+
+  static final linkSmall = GoogleFonts.roboto(
+    color: MyColors.blue,
+    fontSize: 10.sp,
+    fontWeight: FontWeight.w500,
+  );
```
Comment (lukacevica) [inline lib/constants/text_styles.dart:43]: Add new line after.

File: lib/constants/pages.dart [hunk 7]
```diff
@@ -1,22 +1,32 @@
...
   ),
+  GetPage(
+    name: MyRoutes.homeScreen,
+    page: HomeScreen.new,
+    binding: HomeBinding(),
+  ),
 ];
 
-/// All pages have their designated names which can be found here
 class MyRoutes {
```
Comment (lukacevica) [inline lib/constants/pages.dart]: Extract this to separate file.

File: lib/screens/landing/landing_controller.dart [hunk 8]
```diff
@@ -1,29 +1,120 @@
...
+
+  void loginWithEmail() async {
+    final validationCheck = formKey.value.currentState?.validate();
+    if (validationCheck != null && validationCheck) {
+      try {
+        await auth.value.signInWithEmailAndPassword(
+            email: formKey.value.currentState?.fields['email']?.value,
+            password: formKey.value.currentState?.fields['password']?.value);
+      } on FirebaseAuthException catch (e) {
+        if (e.code == 'wrong-password') {
```
Comment (lukacevica) [inline lib/screens/landing/landing_controller.dart]: Make enum with error codes and use switch here.

File: lib/screens/home/home_screen.dart [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    final landingController = Get.put(LandingController());
+
+    return Scaffold(
+        body: Center(
+            child: ListView(
+      children: [
+        Text(FirebaseAuth.instance.currentUser?.email ?? ''),
+        ElevatedButton(
+            onPressed: () => landingController.signOut(),
+            child: const Text('Sign out')),
```
Comment (lukacevica) [inline lib/screens/home/home_screen.dart]: Text not localized.

File: lib/screens/home/home_screen.dart [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+  Widget build(BuildContext context) {
+    //final homeController = Get.put(HomeController());
+
+    final landingController = Get.put(LandingController());
+
+    return Scaffold(
+        body: Center(
+            child: ListView(
+      children: [
+        Text(FirebaseAuth.instance.currentUser?.email ?? ''),
```
Comment (lukacevica) [inline lib/screens/home/home_screen.dart]: For "" make constant empty and call it in cases like this.

File: lib/widgets/shared/texts.dart [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+import 'package:apexology/screens/landing/landing_controller.dart';
+import 'package:apexology/services/connectivity_service.dart';
+import 'package:apexology/widgets/shared/snackbars.dart';
+import 'package:connectivity_plus/connectivity_plus.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:url_launcher/url_launcher_string.dart';
+
+class MyTexts {
```
Comment (lukacevica) [inline lib/widgets/shared/texts.dart]: Don't do it like this, rather make reusable widget with default values.

File: lib/widgets/shared/textfields.dart [hunk 12]
```diff
@@ -0,0 +1,58 @@
+import 'package:apexology/constants/colors.dart';
+import 'package:apexology/screens/landing/landing_controller.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_form_builder/flutter_form_builder.dart';
+import 'package:form_builder_validators/form_builder_validators.dart';
+import 'package:get/get.dart';
+
+class MyTextfields {
```
Comment (lukacevica) [inline lib/widgets/shared/textfields.dart]: Don't do it like this, rather make reusable widget with default values.

File: lib/widgets/shared/buttons.dart [hunk 13]
```diff
@@ -0,0 +1,51 @@
+import 'package:apexology/screens/landing/landing_controller.dart';
+import 'package:get/get.dart';
+import 'package:apexology/services/connectivity_service.dart';
+import 'package:apexology/widgets/shared/snackbars.dart';
+import 'package:connectivity_plus/connectivity_plus.dart';
+import 'package:flutter_signin_button/flutter_signin_button.dart';
+
+class MyButtons {
```
Comment (lukacevica) [inline lib/widgets/shared/buttons.dart]: Don't do it like this, rather make reusable widget with default values.

**Review by lukacevica — CHANGES_REQUESTED** [review]
- Refactor folder structure so that each module has it's own widgets folder with widgets used only in that module
- Make global constants file with numbers e.g. if you have constant corner radius in whole app put it in global constants etc.

**Review by lukacevica — APPROVED** [review]

**Review by PetricMatej — APPROVED** [review]

**Pull request merged**